registration/api/v2/views.py

```diff
@@ -178,6 +181,27 @@ def validate(self, request):
         serializer.is_valid(raise_exception=True)
         return Response(status=status.HTTP_200_OK)
 
+    def create(self, request, *args, **kwargs):
+        """
+        Checks if user is creating account with external domain and fill subdomain slug.
+
+        This check if `external_domain` is filled, if so, generate a valid subdomain slug
+        and fill in the field before passing it to the serializer.
+        """
+        external_domain = request.data.get('external_domain')
+        if external_domain:
+            # Create a valid slug with this format: `<domain-slug>-<random-hash>`
+            # Hash is added to avoid name clashes when generating slug
+            random_hash = ''.join(random.choices(string.ascii_lowercase + string.digits, k=5))
+            subdomain = f"{slugify(external_domain)[:30]}-{random_hash}"
+            # Set request data
+            self.request.data.update({
+                "subdomain": subdomain
+            })
+
+        # Perform create as usual
+        return super(OpenEdXInstanceConfigViewSet, self).create(request, *args, **kwargs)
```

Comment:
@giovannicimolin, should we check for conflict with an existing value and try regenerating the slug again if there is a conflict?

Reply:
It's very unlikely that this will cause any clashes, since the domain name is appended first and we do a check for duplicated external domains. 
The internal domain is the slugified external domain truncated to 32 characters + 5 random digits to make it even harder to have a conflict (for cases with huge domain names where 2 instances have the same first 32 digits).  

Can we merge this without the retry mechanism and see if this causes any issues?